compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/collectors/components/ErrorNodeDiagnosticCollectorComponent.kt

```diff
@@ -54,15 +57,55 @@ class ErrorNodeDiagnosticCollectorComponent(collector: AbstractDiagnosticCollect
         ) return
 
         // If the receiver cannot be resolved, we skip reporting any further problems for this call.
-        if (qualifiedAccess?.dispatchReceiver.hasUnresolvedNameError() || qualifiedAccess?.extensionReceiver.hasUnresolvedNameError() || qualifiedAccess?.explicitReceiver.hasUnresolvedNameError()) return
+        if (qualifiedAccess?.dispatchReceiver.hasUnresolvedNameError() ||
+            qualifiedAccess?.extensionReceiver.hasUnresolvedNameError() ||
+            qualifiedAccess?.explicitReceiver.hasUnresolvedNameError()
+        ) return
+
+        if (reportUninitializedParameter(qualifiedAccess, data)) {
+            return
+        }
 
         reportFirDiagnostic(errorNamedReference.diagnostic, source, reporter, data, qualifiedAccess?.source)
     }
 
     private fun FirExpression?.hasUnresolvedNameError(): Boolean {
-        return when ((this?.typeRef as? FirErrorTypeRef)?.diagnostic) {
-            is ConeUnresolvedNameError -> true
-            else -> false
+        return this?.unresolvedNameError != null
+    }
+
+    private val FirExpression.unresolvedNameError: ConeUnresolvedNameError?
+        get() {
+            val typeRef = if (this is FirAnonymousFunction) this.returnTypeRef else this.typeRef
+            return (typeRef as? FirErrorTypeRef)?.diagnostic as? ConeUnresolvedNameError
+        }
+
+    private fun reportUninitializedParameter(qualifiedAccess: FirQualifiedAccessExpression?, context: CheckerContext): Boolean {
+        if (qualifiedAccess == null) return false
+        val unresolvedNameError = qualifiedAccess.unresolvedNameError ?: return false
+
+        val valueParameter = context.containingDeclarations.lastOrNull { valueParameter ->
+            valueParameter is FirValueParameter &&
+                    valueParameter.defaultValue?.unresolvedNameError == unresolvedNameError
+        } as? FirValueParameter ?: return false
+
+        val function = context.containingDeclarations.lastOrNull { function ->
+            function is FirFunction<*> &&
+                    valueParameter in function.valueParameters &&
+                    unresolvedNameError.name in function.valueParameters.map { it.name }
+        } as? FirFunction<*> ?: return false
+
+        val valueParameterIndex = function.valueParameters.indexOf(valueParameter)
+        val referredParameterIndex = function.valueParameters.map { it.name }.indexOf(unresolvedNameError.name)
```

Comment:
You could get `referredParameterIndex` somehow while you're looking for the matching name (i.e., in line 94). 

Also you can avoid the list created by `map` by doing something like `function.valueParameters.indexOfFirst { it.name == unresolveNameError.name }` instead.